File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>
void makeSpace(int space);
void makeHash(int hash);
void middleSpace();
void newline();
int main(void)
{
    //prompt user for integar input bet 1&8
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);
    //iterate to form pyramid starting from row 1 not zero
    for (int hash = 1; hash <= height; hash++)
    {
        makeSpace(height - hash);
        makeHash(hash);
        middleSpace();
        makeHash(hash);
        newline();
    }

}
//make space first
void makeSpace(int space)
{
    for (int j = 1; j <= space; j++)
    {
        printf(" ");
    }
}
//make hashes
void makeHash(int hash)
{
    for (int z = 1; z <= hash; z++)
    {
        printf("#");
    }
}
// just space between sides
void middleSpace()
{
    printf("  ");
}
//new line after each row
void newline()
{
    printf("\n");
}

File: pset6/mario/more/mario.py
from cs50 import get_int
# prompt the user with +ve integar
height = 0
while height < 1 or height > 8:
    height = get_int("Height: ")
# create half pyramid
for i in range(height):
    # print space
    for space in range(height - i - 1):
        print(" ", end="")
    # print hash
    for ha in range(i+1):
        print("#", end="")
    # print double middle space
    print("  ", end="")
    # print hash
    for ha in range(i+1):
        print("#", end="")
    # print new line
    print("")